crypto/keys/src/x25519.rs

```diff
@@ -419,13 +444,14 @@ impl DistinguishedEncoding for X25519Private {
         Self::try_from(&src[prefix_len..])
     }
 
-    fn to_der(&self) -> Vec<u8> {
-        let mut retval = vec![0u8; X25519_PKI_DER_LEN];
+    fn to_der_slice<'a>(&self, buff: &'a mut [u8]) -> &'a [u8] {
         let key = self.0.to_bytes();
+        buff[..X25519_PKI_DER_LEN].iter_mut().for_each(|b| *b = 0);
+
         let prefix_len = X25519_PKI_DER_PREFIX.len();
-        retval[..prefix_len].copy_from_slice(&X25519_PKI_DER_PREFIX);
-        retval[prefix_len..].copy_from_slice(&key);
-        retval
+        buff[..prefix_len].copy_from_slice(&X25519_PKI_DER_PREFIX);
+        buff[prefix_len..X25519_PKI_DER_LEN].copy_from_slice(&key);
```

Comment:
:wrench: It was missing before, but perhaps this should zeroize `key` since this is a private key
```suggestion
        buff[prefix_len..X25519_PKI_DER_LEN].copy_from_slice(&key);
        key.zeroize();
```